x-pack/test/detection_engine_api_integration/security_and_spaces/group1/fleet_integration.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from 'expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { deleteAllRules, deleteAllTimelines, getPrebuiltRulesStatus } from '../../utils';
+import { deleteAllPrebuiltRules } from '../../utils/delete_all_prebuilt_rules';
+import { installDetectionRulesPackageFromFleet } from '../../utils/install_detection_rules_package_from_fleet';
+import { installPrebuiltRules } from '../../utils/install_prebuilt_rules';
+import { deleteDetectionEnginePackage } from '../../utils/delete_detection_engine_package';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const es = getService('es');
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  describe('fleet_integration', () => {
+    beforeEach(async () => {
+      await deleteDetectionEnginePackage(supertest);
+      await deleteAllRules(supertest, log);
+      await deleteAllTimelines(es);
+      await deleteAllPrebuiltRules(es);
+    });
+
+    /**
+     * Unlike other tests that use mocks, this test uses actual rules from the
+     * package storage and checks that they are installed.
+     */
+    it('should install prebuilt rules from the package storage', async () => {
+      const statusBeforePackageInstallation = await getPrebuiltRulesStatus(supertest);
+      expect(statusBeforePackageInstallation.rules_installed).toBe(0);
+      expect(statusBeforePackageInstallation.rules_not_installed).toBe(0);
+      expect(statusBeforePackageInstallation.rules_not_updated).toBe(0);
+
+      await installDetectionRulesPackageFromFleet({
+        supertest,
+        overrideExistingPackage: true,
+      });
+
+      const statusAfterPackageInstallation = await getPrebuiltRulesStatus(supertest);
+      expect(statusAfterPackageInstallation.rules_installed).toBe(0);
+      expect(statusAfterPackageInstallation.rules_not_installed).toBeGreaterThan(0);
```

Comment:
Should we specify some threshold-like positive number here, e.g. 500? We know that we have 700+ rules now, and this threshold could theoretically catch a bug in the package where for some reason a lot of rules end up being not shipped. On the other hand, that's probably not a problem with this endpoint, so I'm not sure. What do you think?

Reply:
It's not really the rules installation endpoint's responsibility to check the rules package content. For example, we could split one monolithic package into multiple smaller packages in the future, and the endpoint should continue working without any changes. So, I don't think we need to worry about verifying it on our side. The TRaDE team is actually in charge of testing the package content, and they do a bunch of tests before every rule release. So I think we should be good there.